application/core/LSYii_Application.php

```diff
@@ -717,4 +717,93 @@ public function getPublicBaseUrl($absolute = false)
         }
         return $baseUrl;
     }
+
+	/**
+	 * Is REST request
+	 *
+	 * @return boolean
+	 */
+    public function isRestRequest()
+    {
+        $headers = getallheaders();
+        $acceptsJson = isset($headers['Accept'])
+            && strpos($headers['Accept'], 'application/json') !== false;
+
+        $isRestUrl = !isset($_SERVER) ||
+            !isset($_SERVER['REQUEST_URI']) ||
+            strstr($_SERVER['REQUEST_URI'], '/rest/');
+
+        return $acceptsJson || $isRestUrl;
+    }
+
+	/**
+	 * Handles uncaught PHP exceptions.
+	 *
+	 * @param Exception $exception exception that is not caught
+	 */
+    public function handleException($exception)
```

Comment:
Hm isn't this configured in config.php? https://www.yiiframework.com/doc/guide/1.1/en/topics.error
Or you can't branch on that point, only on runtime?

Reply:
I noticed we have custom error handling code in the onException() method. I have moved the error handling code in to a separate class and added the extra functionality.